### ISSUE TITLE
Target in arc not recognized [Bug]

### ISSUE THREAD

**Issue by Sergovan1** [Sandrem/FlyCasual#2679]
I have 2 instances of the arc line passing through a base and it not being recognized. 

I don't know if it is  as a result of it being a medium base attacker or what not, but the first one is against an E-wing (cuts through the corner) and the 2nd one it cuts 1/4 of the base if the TIE fighter. In each case, I put a sheet of paper on the screen and zoomed in to measure it. I have often found that ones that looked in were in fact out by a hair so I was a little surprised to see these 2 cases were considered out.
![Screen Shot 2020-05-22 at 7 01 19 PM](https://user-images.githubusercontent.com/64314306/82853989-2be6f380-9ed5-11ea-9a18-05cfeacf05c7.png)
![Screen Shot 2020-05-24 at 9 09 05 AM](https://user-images.githubusercontent.com/64314306/82853993-2db0b700-9ed5-11ea-9fb0-502f6965b825.png)

**Comment by Sandrem** [comment]
Please, if you will see this again - stop any actions, go to "%userprofile%\AppData\LocalLow\Sandrem\Fly Casual\Second Edition\Replays\" and attach that replay here, with information from what to what ship do you measure.

I will try to play replay and check arc detection by myself.

**Comment by Sergovan1** [comment]
I just had this issue come up. Question, how do I get the replay file if I am on a Mac computer?

Here you can see my Arc shooting at another Arc. They are mid range 3 and over half the base is in arc, yet the target (in red) is not recognized as valid for attacking.
 
![Screen Shot 2020-05-26 at 8 52 19 PM](https://user-images.githubusercontent.com/64314306/82964468-89417a00-9f93-11ea-97f1-18a835d86c4b.png)

**Comment by Sandrem** [comment]
@Sergovan1 Try "/Library/Application Support/Sandrem/Fly Casual/Second Edition/Replays"

**Comment by Sergovan1** [comment]
I can open contents of FlyCasual. I get a folder showing "Contents". In that I get Frameworks, MacOS, MonoBleedingEdge, and Resource folders. I have looked through all of these sub folders and I can't find any file called Replay.

**Comment by Sandrem** [comment]
No, this folder is not in files that you run - it is in MacOS' folders, where user's settings/data is stored.
https://nektony.com/blog/how-to-get-to-library-on-mac

**Comment by Sergovan1** [comment]
Ok, I think I got my last replay. Check Jag to his target (as per screenshot) above. Let me know if this works.

[LastReplay.replay.zip](https://github.com/Sandrem/FlyCasual/files/4691310/LastReplay.replay.zip)

**Comment by reaper802** [comment]
Same issue, twice in the same game.

![image](https://user-images.githubusercontent.com/65979002/83193358-de50cd80-a104-11ea-82f5-86d9b29d8d06.png)

![image](https://user-images.githubusercontent.com/65979002/83193397-f294ca80-a104-11ea-9f92-7df208af6fbe.png)

![image](https://user-images.githubusercontent.com/65979002/83193408-f88aab80-a104-11ea-881e-7d081d7ceda1.png)

**Comment by Sergovan1** [comment]
It is really hard to see the line in 3D. In your 2D screenshot, if you put a piece of paper on the arc edge you will see that the enemy base is not in arc. The range measurement is from corner to corner for range. It is not an in arc check. If you line up a straight edge to the screenshots I posted, you will see how those do intersect the targets base.

**Comment by whittaker007** [comment]
Probably the same issue here, came up a couple of times in the last game. 

Not 100% sure but it looks like it's measuring closest corner to closest corner and finding it's not in arc, but the closest point _in arc_ is _not_ the corner, and is in range:

<img width="665" alt="Screen Shot 2020-06-23 at 11 16 09 AM" src="https://user-images.githubusercontent.com/1250821/85345700-4c8f8100-b547-11ea-8457-ae5541082bc6.png">

If this is not the same issue I can create a new one?

**Comment by whittaker007** [comment]
Here's another example: Cova is clearly range 1 in arc of the Arc, yet it thinks that it is not in arc at all.

<img width="848" alt="Screen Shot 2020-07-11 at 10 13 57 AM" src="https://user-images.githubusercontent.com/1250821/87208739-bb504500-c363-11ea-8e50-60849e6f5039.png">
<img width="612" alt="Screen Shot 2020-07-11 at 10 20 20 AM" src="https://user-images.githubusercontent.com/1250821/87209075-ce174980-c364-11ea-9909-04173789fbe6.png">

**Comment by whittaker007** [comment]
If I can offer a suggestion, I'd do away with using the simulated range ruler entirely and use a straight line like a laser beam. That would make it easier to see visually. Projected arcs with range bands would also be a nice feature.

**Comment by Solomanus** [comment]
![Dengar_Left_Not-in-arc](https://user-images.githubusercontent.com/50461461/113220357-e8b58980-9237-11eb-826c-b9a9c2e6d074.png)

Clearly in arc at range 1, but showing as not in left arc. (Replay available, if necessary.)

**Comment by whittaker007** [comment]
Here's another example. Clearly in arc and at range 1, but it says not in arc at all. It seems very clear that the problem is that it is comparing the closest point to closest point and determining that it is not in arc. 

<img width="591" alt="Screen Shot 2021-04-04 at 12 35 15 PM" src="https://user-images.githubusercontent.com/1250821/113495226-2a9a4480-9544-11eb-91bc-626f507a6f28.png">

[LastReplay.replay.zip](https://github.com/Sandrem/FlyCasual/files/6253822/LastReplay.replay.zip)

Closest point to closest point should only be used for determining the relative range of one object to another. A separate calculation is needed to determine if the target is in the attack arc at all, and another to find the closest point within that arc, and finally determine what range that point is at.

**Comment by Sandrem** [comment]
@whittaker007 In your example - enemy ship is not in front arc.
![Not in arc](https://user-images.githubusercontent.com/6347443/113509534-d8b2e800-955e-11eb-88e9-510907ccba4e.jpg)

**Comment by whittaker007** [comment]
Oh wow, my apologies. This must be why I'm so bad at X-Wing on the tabletop! 

Thanks,
Scott

**Comment by whittaker007** [comment]
Sorry, but how about this one:

[LastReplay.replay.zip](https://github.com/Sandrem/FlyCasual/files/6256806/LastReplay.replay.zip)
<img width="223" alt="Screen Shot 2021-04-05 at 6 57 39 PM" src="https://user-images.githubusercontent.com/1250821/113547427-002eb100-9642-11eb-9c03-c42fd41b032f.png">
